app/services/inclusion_connect.rb

```diff
@@ -64,28 +64,59 @@ def get_user_info(token)
     JSON.parse(res.body)
   end
 
-  def update_agent(agent, user_info)
-    agent.inclusion_connect_open_id_sub ||= user_info["sub"]
-    agent.first_name ||= user_info["given_name"]
-    agent.last_name ||= user_info["family_name"]
+  def update_agent(agent)
+    # We dont want to update one of the agents if we have a mismatch
+    return handle_agent_mismatch if agent_mismatch?
 
-    agent.invitation_accepted_at ||= Time.zone.now
-    agent.confirmed_at ||= Time.zone.now
+    update_basic_info(agent)
+    update_email_and_uid(agent) if agent.email != @user_info["email"]
+  end
 
-    agent.last_sign_in_at = Time.zone.now
-    agent.save!
+  def update_basic_info(agent)
+    agent.assign_attributes({
+                              inclusion_connect_open_id_sub: agent.inclusion_connect_open_id_sub || @user_info["sub"],
+                              first_name: @user_info["given_name"],
+                              last_name: @user_info["family_name"],
+                              invitation_accepted_at: agent.invitation_accepted_at || Time.zone.now,
+                              # Setting the token to nil to disable old invitations links
+                              invitation_token: nil,
+                              confirmed_at: agent.confirmed_at || Time.zone.now,
+                              last_sign_in_at: Time.zone.now,
+                            })
```

Comment:
Que dis-tu de cette indentation qui est plus en continuité avec la forme du code environnant ? :upside_down_face: 

```suggestion
    agent.assign_attributes(
      {
        inclusion_connect_open_id_sub: agent.inclusion_connect_open_id_sub || @user_info["sub"],
        first_name: @user_info["given_name"],
        last_name: @user_info["family_name"],
        invitation_accepted_at: agent.invitation_accepted_at || Time.zone.now,
        # Setting the token to nil to disable old invitations links
        invitation_token: nil,
        confirmed_at: agent.confirmed_at || Time.zone.now,
        last_sign_in_at: Time.zone.now,
      }
    )
```